### PR TITLE
Documents some Timer peripherals

### PR DESCRIPTION
**Pull request by octronics**
Describes some common fields on basic TIM.
Describes TIM16 peripheral (apply on TIM17).
Describes TIM6 peripheral (apply on TIM7).
Add TIM16 to all devices that has this hardware.
Add TIM6 to all devices that has this hardware.
Add basic_timer on the STM32F0 family.
Describes some fields of TIM2 peripheral.
Add 32bit TIM2 peripheral to all devices that has it (all except F1 and L1).
Add 16bit TIM2 peripheral to all devices that has it (F1 and L1 family).

### CODE REVIEW

**Comment by octronics** [comment]
More 32bit TIM2 fields to come. But not for today. You can merge as it, if you want.

**Comment by adamgreig** [comment]
Looks like the Travis failure was spurious, I've rerun it and it's OK now.

Are you finished with this PR (is it ready to review and merge)?

**Comment by octronics** [comment]
I have more to come, but lets those be reviewed first. I'm not sure about value names. Could you take a look at them?

**Comment by octronics** [comment]
Rename to `Enabled` and `Disabled` as used everywhere else. The only usage of `Enable` as on `iwdg` and `wwdg` and comes from my previous pull request. I will correct them later.

File: peripherals/tim/tim16.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+    ARPE:
+      NotBuffered: [0, "TIMx_ARR register is not buffered"]
+      Buffered: [1, "TIMx_ARR register is buffered"]
+  CR2:
+    CCPC:
+      NotPreloaded: [0, "CCxE, CCxNE and OCxM bits are not preloaded"]
+      Preloaded: [1, "CCxE, CCxNE and OCxM bits are preloaded"]
+    CCUS:
+      WithoutRisingEdge: [0, "Capture/compare are updated only by setting the COMG bit"]
+      WithRisingEdge: [1, "Capture/compare are updated by setting the COMG bit or when an rising edge occurs on TRGI"]
```
Comment (octronics) [inline peripherals/tim/tim16.yaml]: Is `WithoutRisigngEdge` good or something like `Default` better?

File: peripherals/tim/tim2_common.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+      Enable: [1, "The counter enable signal, CNT_EN, is used as trigger output"]
+      Update: [2, "The update event is selected as trigger output"]
+      ComparePulse: [3, "The trigger output send a positive pulse when the CC1IF flag it to be set, as soon as a capture or a compare match occurred"]
+      CompareOC1: [4, "OC1REF signal is used as trigger output"]
+      CompareOC2: [5, "OC2REF signal is used as trigger output"] 
+      CompareOC3: [6, "OC3REF signal is used as trigger output"]
+      CompareOC4: [7, "OC4REF signal is used as trigger output"]
+    CCDS:
+      OnCCx: [0, "CCx DMA request sent when CCx event occurs"]
+      OnUpdate: [1, "CCx DMA request sent when update event occurs"]
```
Comment (octronics) [inline peripherals/tim/tim2_common.yaml]: Perhaps is `OnCompare` better than `OnCCx`?

File: peripherals/tim/tim16.yaml (hunk 1 above)
Comment (octronics) [inline peripherals/tim/tim16.yaml]: Is WithoutRisigngEdge good or something like Default better?

File: peripherals/tim/tim2_common.yaml (hunk 2 above)
Comment (octronics) [inline peripherals/tim/tim2_common.yaml]: Perhaps is OnCompare better than OnCCx?

Comment (adamgreig) [reply peripherals/tim/tim2_common.yaml]: I agree, `OnCompare` seems nicer.

File: peripherals/tim/tim16.yaml (hunk 1 above)
Comment (adamgreig) [reply peripherals/tim/tim16.yaml]: I don't mind either way here, your choice.

File: peripherals/tim/tim2_common.yaml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+# Common base for 16bit and 32bit TIM2 peripheral
+
+"TIM[23]":
+  CR1:
+    CKD:
+      NotDivided: [0, "CK_INT not divided"]
+      DividedBy2: [1, "CK_INT divided by 2"]
+      DividedBy4: [2, "CK_INT divided by 4"]
+    CMS:
+      Edge: [0, "The counter counts up or down depending on th direction bit (DIR)"]
```
Comment (adamgreig) [inline peripherals/tim/tim2_common.yaml]: s/th/the

**Comment by adamgreig** [comment]
Great, thanks!

**Pull request merged**